docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,492 @@
+    - [Patch the cluster to use the new OIDC](#patch-cluster)
+    - [Revoke public access to S3 Bucket](#revoke-s3-public-access)
+- [Rollback to OIDC with S3 Public URL](#rollback)
+- [Delete the Old OIDC identity provider](#delete)
+
+## Prerequisites <a name="prerequisites"></a>
+
+1) An OCP Cluster created on AWS with manual mode with STS using S3 Bucket as OIDC URL (issuerURL)
+
+> **NOTE**: The steps described in this document was tested and is validated to patch IAM Roles created by **cluster components**. **You must patch the IAM Role Trusted Policies [created to user-workload by pod identity webhook steps](https://docs.openshift.com/rosa/authentication/assuming-an-aws-iam-role-for-a-service-account.html).
+
+2) Export the environment variables below - required to be logged on OCP:
+```bash
+# Discovery the `--name` provided when creating resources by CCO
+export OIDC_BUCKET_HOST=$(basename $(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} ))
+export OIDC_ARN_S3=$(aws iam list-open-id-connect-providers | jq -r ".OpenIDConnectProviderList[] | select(.Arn | endswith(\"$OIDC_BUCKET_HOST\")).Arn")
+export CLUSTER_NAME=$(aws iam list-open-id-connect-provider-tags --open-id-connect-provider-arn $OIDC_ARN_S3 | jq -r '.Tags[] | select(.Key=="Name").Value')
+export OIDC_BUCKET_NAME=$(echo $OIDC_BUCKET_HOST | awk -F'.' '{print$1}')
+export CLUSTER_REGION=$(echo $OIDC_BUCKET_HOST | awk -F'.' '{print$3}')
+```
+
+Check the discovered var values
+
+> All the values must be discovered.
+
+```bash
+cat <<EOF
+OIDC_BUCKET_HOST=$OIDC_BUCKET_HOST
+OIDC_ARN_S3=$OIDC_ARN_S3
+CLUSTER_NAME=$CLUSTER_NAME
+OIDC_BUCKET_NAME=$OIDC_BUCKET_NAME
+CLUSTER_REGION=$CLUSTER_REGION
+EOF
+```
+
+3) Make sure you can reach (read) the S3 Bucket created with the default name
+
+```bash
+aws s3 ls --region $CLUSTER_REGION s3://$OIDC_BUCKET_NAME
+```
+
+4) A clean work directory: a lot of files will be created, make sure you switched to a new work directory to save the files properly (it can be used in the future for rollback)
+
+5) An OpenShift user with cluster-admin permissions
+
+6) AWS policies required by the agent running that steps:
+
+```JSON
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "Stmt1678336730279",
+      "Action": [
+        "iam:CreateOpenIDConnectProvider",
+        "iam:DeleteOpenIDConnectProvider",
+        "iam:GetRole",
+        "iam:ListOpenIDConnectProviderTags",
+        "iam:ListOpenIDConnectProviders",
+        "iam:ListRoles",
+        "iam:UpdateAssumeRolePolicy",
+        "cloudfront:CreateDistribution",
+        "cloudfront:CreateCloudFrontOriginAccessIdentity"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Sid": "Stmt1678336953197",
+      "Action": [
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:PutObject",
+        "s3:PutPublicAccessBlock"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
```

Comment:
Here is the updated json after my testing.

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1678336730279",
            "Action": [
                "iam:CreateOpenIDConnectProvider",
                "iam:DeleteOpenIDConnectProvider",
                "iam:GetRole",
                "iam:ListOpenIDConnectProviderTags",
                "iam:ListOpenIDConnectProviders",
                "iam:ListRoles",
                "iam:UpdateAssumeRolePolicy",
                "cloudfront:CreateDistribution",
                "cloudfront:CreateCloudFrontOriginAccessIdentity",
                "cloudfront:TagResource"
            ],
            "Effect": "Allow",
            "Resource": "*"
        },
        {
            "Sid": "Stmt1678336953197",
            "Action": [
                "s3:GetObject",
                "s3:ListBucket",
                "s3:PutObject",
                "s3:PutBucketPublicAccessBlock",
                "s3:PutBucketPolicy"
            ],
            "Effect": "Allow",
            "Resource": "*"
        }
    ]
}